src/Compiler/CompilerPasses.cpp

```diff
@@ -60,6 +60,12 @@ void addONNXToMLIRPasses(mlir::PassManager &pm, bool targetCPU) {
 
   pm.addNestedPass<func::FuncOp>(onnx_mlir::createDecomposeONNXToONNXPass());
   pm.addPass(onnx_mlir::createShapeInferencePass());
+
+  /* FlexML Start */
+  if (layerNameToLocation)
```

Comment:
Does something like `onnx-mlir -EmitONNXBasic | onnx-mlir-opt -layer-name-to-location | onnx-mlir -emitONNXIR` also work? I would prefer that over further modifying the core onnx-mlir driver.

Reply:
That would require creating 2 more pipeline steps in our driver. I'm not a big fan of that, since we also cannot pipe in our driver and instead need to create even more additional files (which we don't check in, sure, but still).

Reply:
It could still be in the same python pipeline step and we could immediately delete the old file. But I also see your point. It would just smoothen the way to get rid of the forks... Maybe it is to early to think about that. But you could add the command as a comment so that we have a hint later on